lib/problems.rb

```diff
@@ -1,13 +1,32 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  return false if string.length % 2 != 0
+  
+  struct = Stack.new
+  left_half = ["(", "[", "{"]
+  right_half = [")", "]", "}"]
```

Comment:
Did you know, you could also make a hash where the keys are the open parens and the values are the close parens.  That way you wouldn't have to check the indexes like you are.